### ISSUE TITLE
Stun Events in Devastation Sims Crash

### ISSUE THREAD

**Issue by ChrisBrown46** [simulationcraft/simc#7849]
### Describe the bug

When adding custom `stun` raid events via 
`raid_events+=/stun,cooldown=999,duration=30,first=100,last=130`

the simulation will fail with the error
`Error in child simulation (x): Player '...' scheduled ready while already ready.`

or with the error
`Error in child simulation (x): Non-channeling action eternity_surge_damage for Player '...' is trying to overwrite player-ready-event upon execute.`

Example via Raidbots inside the `SimC Notifications` section: <https://www.raidbots.com/simbot/report/pWSd6Yh8Cm5taSqtf8ZrXR>

When this error occurs, the simulation stops which leaves the iterations ~20 in total.

### Expected behavior

Testing the same simulation options but with a different profile, such as a shadow priest, produces expected results with no errors. Example on Raidbots: <https://www.raidbots.com/simbot/report/pUDwJ9TMSVXCffjAV9uAGL>

### To Reproduce

To reproduce, use the nightly SimC version (did not personally test older versions) and import a Devastation Evoker profile, such as Lunchbox on Thrall. Testing with and without specific talents did not fix the issue, so any talents should work.

Add to the header any custom stun events, such as
```
# stun timings (underworld) 
raid_events+=/stun,cooldown=31,duration=30,first=100,last=130 
raid_events+=/stun,cooldown=31,duration=30,first=230,last=260 
raid_events+=/stun,cooldown=21,duration=20,first=400,last=420
```
and simulate a normal 10 minute patchwerk fight, or shorter so long as each stun event will occur in the given duration. Each stun event will cause an error, so three stun events will create three errors.

### Additional information
* SimulationCraft Version: 1010-01

* No SimC hard crashes.

* **simc_gui.simc** file or contents
  <https://pastebin.com/pTQiK0AC>

**Comment by seanpeters86** [comment]
CC: @Saeldur @gastank

**Comment by gastank** [comment]
should be fixed in 876ceb5e6b

re-open if something else still goes awry
